### PR TITLE
FileOpenPicker, FileSavePicker on Skia WPF, collections, common properties

### LINKED ISSUE #5234: [Skia WPF] FileSavePicker
Part of #3215

## What would you like to be added:

`FileSavePicker` on Skia WPF.

## Why is this needed:

💾

## For which Platform:

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [x] Skia
  - [x] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### LINKED ISSUE #5235: [Skia WPF] FileOpenPicker
Part of #508

## What would you like to be added:

`FileOpenPicker` API on Skia WPF

## Why is this needed:

UWP parity

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [x] WPF
  - [ ] GTK (Linux)
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### LINKED ISSUE #5239: Add support for Storage.Pickers collections
## What would you like to be added:

Various collections under `Windows.Storage.Pickers`

## Why is this needed:

Full support for pickers.

## Notes

**Requirements**:
- [x] Ensure validation on `Add`
- [x] Ensure `null` validation
- [x] Ensure wildcard support
- [x] Ensure duplicate validation

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): related to #3916, #508, #3215, closes #5234, closes #5235, closes #5239

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

- File picker collections are not supported
- Some properties throw

## What is the new behavior?

- File picker collection are implemented
- Common properties are implemented

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/Storage/Pickers/FolderPicker.macOS.cs [hunk 1]
```diff
@@ -1,38 +1,32 @@
...
+	{
+		private async Task<StorageFolder?> PickSingleFolderTaskAsync(CancellationToken token)
+		{
+			var openPanel = new NSOpenPanel();
+			openPanel.AllowedFileTypes = new[] { "none" };
+			openPanel.AllowsOtherFileTypes = false;
+			openPanel.CanChooseDirectories = true;
+			openPanel.CanChooseFiles = false;
+			var response = openPanel.RunModal();
+			if (response == 1 && openPanel.Urls.Length > 0)
```
Comment (ajpinedam) [inline src/Uno.UWP/Storage/Pickers/FolderPicker.macOS.cs:21]: Didn't understand the `1` but I see that Xamarin hasn't yet mapped the `struct` returned from this method in the Apple implementation `NSApplication.ModalResponse`

**Review by ajpinedam — APPROVED** [review]
Beautiful code!

File: src/Uno.UWP/Storage/Pickers/FolderPicker.skia.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    {
+		private IFolderPickerExtension? _folderPickerExtension;
+
+		partial void InitializePlatform() => ApiExtensibility.CreateInstance(this, out _folderPickerExtension);
+
+		private async Task<StorageFolder?> PickSingleFolderTaskAsync(CancellationToken token)
+		{
+			if (_folderPickerExtension == null)
+			{
+				throw new NotSupportedException("FileOpenPicker is not supported on this target.");
```
Comment (carldebilly) [inline src/Uno.UWP/Storage/Pickers/FolderPicker.skia.cs:21]: Maybe the name "FileOpenPicker" is not correct here ;-)

File: src/Uno.UWP/Storage/Pickers/FolderPicker.macOS.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UWP/Storage/Pickers/FolderPicker.macOS.cs]: I have moved it into a constant at least :-) .

File: src/Uno.UWP/Storage/Pickers/FolderPicker.skia.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UWP/Storage/Pickers/FolderPicker.skia.cs]: Fixd :-)

**Review by carldebilly — APPROVED** [review]

**Pull request merged**